pkg/api/v1/status/atlascluster.go

```diff
@@ -0,0 +1,69 @@
+package status
+
+import "go.mongodb.org/atlas/mongodbatlas"
+
+// AtlasClusterStatus defines the observed state of AtlasCluster.
+type AtlasClusterStatus struct {
+	Common `json:",inline"`
+
+	// StateName is the current state of the cluster.
+	// The possible states are: IDLE, CREATING, UPDATING, DELETING, DELETED, REPAIRING
+	// +optional
+	StateName string `json:"stateName,omitempty"`
+
+	// MongoDBVersion is the version of MongoDB the cluster runs, in <major version>.<minor version> format.
+	// +optional
+	MongoDBVersion string `json:"mongoDBVersion,omitempty"`
+
+	// ConnectionStrings is a set of connection strings that your applications use to connect to this cluster.
+	// +optional
+	ConnectionStrings *ConnectionStrings `json:"connectionStrings"`
+
+	// MongoURIUpdated is a timestamp in ISO 8601 date and time format in UTC when the connection string was last updated.
+	// The connection string changes if you update any of the other values.
+	// +optional
+	MongoURIUpdated string `json:"mongoURIUpdated,omitempty"`
+}
+
+// ConnectionStrings is a copy of mongodbatlas.ConnectionStrings for deepcopy compatibility purposes.
+type ConnectionStrings struct {
```

Comment:
can we have descriptions for these fields as well? And generate the CRD after that